### PR TITLE
Initialize log4j with log4j2.xml

### PR DESCRIPTION
**Pull request by djuarezgf**
**What's in the PR**
* Hotfix: Initialize log4j config with log4j2.xml

### CODE REVIEW

File: src/main/java/de/samply/share/client/listener/StartupListener.java [hunk 1]
```diff
@@ -33,9 +34,19 @@
 
   @Override
   public void contextInitialized(ServletContextEvent sce) {
-    logger.info("context initialized!");
     setMessagesResolver();
     ProjectInfo.INSTANCE.initProjectMetadata(sce);
+    try {
```
Comment (djuarezgf) [inline src/main/java/de/samply/share/client/listener/StartupListener.java:39]: We recover this old code in order to read the log4j2.xml in the right place.

File: pom.xml [hunk 2]
```diff
@@ -287,6 +287,11 @@
       <artifactId>log4j-core</artifactId>
       <version>${log4j.version}</version>
     </dependency>
+    <dependency>
```
Comment (djuarezgf) [inline pom.xml:290]: Needed in order to use in log4j2.xml the lookup ${web:servletContextName}

File: src/main/java/de/samply/share/client/listener/StartupListener.java [hunk 3]
```diff
@@ -33,9 +34,19 @@
...
+      Configurator.initialize(
+          null,
+          FileFinderUtil.findFile(
+              "log4j2.xml", ProjectInfo.INSTANCE.getProjectName(),
+              System.getProperty("catalina.base") + File.separator + "conf",
+              sce.getServletContext().getRealPath("/WEB-INF")).getAbsolutePath());
+      logger.info("context initialized!");
+    } catch (FileNotFoundException e) {
+      e.printStackTrace();
```
Comment (alexanderkiel) [inline src/main/java/de/samply/share/client/listener/StartupListener.java:48]: Please use logger.error here.

**Review by alexanderkiel — CHANGES_REQUESTED** [review]

File: src/main/java/de/samply/share/client/listener/StartupListener.java (hunk 3 above)
Comment (djuarezgf) [reply src/main/java/de/samply/share/client/listener/StartupListener.java]: done

**Review by alexanderkiel — APPROVED** [review]

**Review by alexanderkiel — APPROVED** [review]

**Pull request merged**